### ISSUE TITLE
[Rancher] No IP in backend in host networking mode

### ISSUE THREAD

**Issue by tboyce021** [containous/traefik#1498]
### Do you want to request a *feature* or report a *bug*?
Bug. Appears to be the same as #487 except for Rancher (I even ran into it setting up Home Assistant as well).

### What did you do?
Set up Traefik with Rancher backend for a container with net=host.

### What did you expect to see?
IP address in backend

### What did you see instead?
IP address is missing

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.2.3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
# (paste your configuration here)
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
(paste your output here)
```

**Comment by ldez** [comment]
Could you provide more information? (step to reproduce, logs, configuration, ...)

**Comment by dniel** [comment]
I have seen the same issue when debugging #1491, no ip registered when setting the service with [network mode to HOST](http://docs.rancher.com/rancher/v1.5/en/rancher-services/networking/#host), only port number registered.

**Comment by tboyce021** [comment]
It's the exact same issue as #487 except for the Rancher backend instead of Docker.

To reproduce: Just set up Traefik with the Rancher backend and set one of the exposed containers/services to `Host` network mode. The Traefik backend will show the URL as `http://<port>`.

This is because Rancher provides a null IP address for containers in host mode. You can see this by just viewing the container in the Rancher API and you'll see `primaryIpAddress": null`. This is then used here: https://github.com/containous/traefik/blob/master/provider/rancher/rancher.go#L399.

The Docker workaround in #494 will work if the container is running on the same machine as Traefik, but otherwise it won't (same as with Docker swarm mode). The better solution would be to use the IP address of the container's host.

**Comment by timoreimann** [comment]
This sounds somewhat similar to the Marathon-specific problem #1243, which got solved by auto-detecting the right mode (#1345).

**Comment by dniel** [comment]
Yes this sounds very similar to #1243 .
I'm trying to find a workaround in my project, but still from what I figured out, the rancher provider does not support traefik as an external load balancer for services deployed by Rancher because of the ip of the services is always reported either as the internal rancher network address or missing.

Right now I'm trying to figure out how to customize the template for rancher provider for a custom workaround..

**Comment by matq007** [comment]
Any update on how to solve this issue? I would like to use Traefik as a reverse-proxy on rancher :)

**Comment by traefiker** [comment]
Closed by #2101.
